src/renderer/hooks/use-active-heading.ts

```
@@ -0,0 +1,133 @@
+      }
+    )
+
+    const elements = Array.from(
+      container.querySelectorAll<HTMLElement>('[data-node-type="blockContainer"][data-id]')
+    ).filter((element) => {
+      const elementId = element.dataset.id
+      const isHeadingBlock = !!element.querySelector('[data-content-type="heading"]')
+      return !!elementId && isHeadingBlock && headingIds.has(elementId)
+    })
+
+    elements.forEach((element) => observer.observe(element))
```

Biome (2.4.4)

[error] This callback passed to forEach() iterable method should not return a value.

(lint/suspicious/useIterableCallbackReturn)